src/lib/image/open/open.c

```diff
@@ -60,12 +60,10 @@ int _singularity_image_open(struct image_object *image, int open_flags) {
     }
 
     singularity_message(DEBUG, "Opening file descriptor to image: %s\n", image->path);
-    if ( ( image->fd = open(image->path, open_flags, 0755) ) < 0 ) {
+    if ( ( image->fd = open(image->path, open_flags|O_CLOEXEC, 0755) ) < 0 ) {
```

Comment:
@cclerget, we have had problems in the past with SuSE not supporting this option to open(). Instead could we use fcntl()?